lib/web.ts

```
@@ -28,8 +28,10 @@ function filterCMPs(rules: AutoCMP[], config: Config) {
 export default class AutoConsent {
     id = getRandomID();
     rules: AutoCMP[] = [];
+    // @ts-expect-error - config is initialized in initialize
     config: Config;
-    foundCmp: AutoCMP = null;
+    // @ts-expect-error - foundCmp is initialized in findCmp
+    foundCmp: AutoCMP;
```

TypeScript is correct. If `config` is not given to the constructor then these fields aren’t initialised. We shouldn’t silence the error.

---

That’s true that `config` is always initialised after `initialize()`, but it’s not always initialised after `constructor()`, which is the problem TypeScript is (correctly) identifying. 

The reality is that `config` will be undefined until `initialize()` has been called whether that’s when class is instantiated or when `initResp` is received. The types should reflect reality.

Perhaps make `config: Config | undefined` private and then have a `config: Config` that throws if `initialize()` hasn’t been called yet?

```diff
diff --git a/lib/web.ts b/lib/web.ts
index 4802d0d..0559cd3 100644
--- a/lib/web.ts
+++ b/lib/web.ts
@@ -28,8 +28,7 @@ function filterCMPs(rules: AutoCMP[], config: Config) {
 export default class AutoConsent {
     id = getRandomID();
     rules: AutoCMP[] = [];
-    // @ts-expect-error - config is initialized in initialize
-    config: Config;
+    #config?: Config;
     foundCmp?: AutoCMP;
     state: ConsentState = {
         cosmeticFiltersOn: false,
@@ -76,7 +75,7 @@ export default class AutoConsent {
     initialize(config: Partial<Config>, declarativeRules: RuleBundle | null) {
         const normalizedConfig = normalizeConfig(config);
         normalizedConfig.logs.lifecycle && console.log('autoconsent init', window.location.href);
-        this.config = normalizedConfig;
+        this.#config = normalizedConfig;
         if (!normalizedConfig.enabled) {
             normalizedConfig.logs.lifecycle && console.log('autoconsent is disabled');
             return;
@@ -131,8 +130,15 @@ export default class AutoConsent {
         this.updateState({ lifecycle: 'initialized' });
     }
 
+    get config() {
+        if (!this.#config) {
+            throw new Error('AutoConsent is not initialized yet');
+        }
+        return this.#config;
+    }
+
     get shouldPrehide() {
-        return this.config?.enablePrehide && !this.config?.visualTest;
+        return this.config.enablePrehide && !this.config.visualTest;
     }
 
     saveFocus() {
```